tests/ir/test_gate.py

```diff
@@ -137,3 +137,37 @@ def test_pickle(self, gate: Gate) -> None:
         utry = gate.get_unitary(params)
         pickled_utry = pickle.loads(pickle.dumps(gate)).get_unitary(params)
         assert utry == pickled_utry
+
+    def test_get_inverse(self, gate: Gate) -> None:
+        from bqskit.ir.gates.constantgate import ConstantGate
+        from bqskit.qis.unitary.differentiable import DifferentiableUnitary
+        from bqskit.ir.gates.qubitgate import QubitGate
+        from bqskit.ir.operation import Operation
+        if isinstance(gate, QubitGate):
+            if isinstance(gate, ConstantGate):
+                assert hasattr(gate, 'get_inverse')
+                inv_gate = gate.get_inverse()
+                assert inv_gate._qasm_name, \
+                    'All inverses should have an effective _qasm_name'
+                iden = np.identity(gate.dim)
+                inv_gate = gate.get_inverse()
+                supposed_to_be_iden = (
+                    inv_gate.get_unitary() @ gate.get_unitary()
+                )
+                dist = supposed_to_be_iden.get_distance_from(iden, 1)
```

Comment:
Is `get_distance_from` the correct way to test here? This will ignore global phase, I have to think we should allow this. If you make the test more strict and just check the norm to identity, for example, do you have any issues?

Also, we should test multiplication both ways, A@B=I and B@A=I.